kraken/invoke/command.py

```diff
@@ -5,8 +5,17 @@
 # Invokes a given command and returns the stdout
 def invoke(command):
     try:
-        output = subprocess.check_output(command, shell=True,
-                                         universal_newlines=True)
+        output = subprocess.Popen(command, shell=True,
+                                  universal_newlines=True, stdout=subprocess.PIPE,
+                                  stderr=subprocess.STDOUT)
+        (out, err) = output.communicate()
+    except Exception as e:
+        logging.error("Failed to run %s, error: %s" % (command, e))
+    return out
+
+
+def run(command):
+    try:
+        subprocess.run(command, shell=True, universal_newlines=True, timeout=45)
     except Exception:
-        logging.error("Failed to run %s" % (command))
```

Comment:
I think you should keep a logging statement here and not just pass

Reply:
This function was added just for the stop_kubelet node scenario, when this command is executed once, the kubelet stops even if the command exceeds the timeout (here set to 45s) and exception indicating 'failed to run command' wouldn't be meaningful in that case.